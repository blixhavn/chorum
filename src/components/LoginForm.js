import React, { Component } from 'react';
import { View, Text, Image, TouchableHighlight, StyleSheet, TextInput } from 'react-native';
import { connect } from 'react-redux';
import { Actions } from 'react-native-router-flux';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { faAt, faKey } from '@fortawesome/free-solid-svg-icons';
import { emailChanged, passwordChanged, loginUser } from '../actions';
import { Button, Spinner } from './common';
import { colors } from './common/constants';

export class LoginForm extends Component {

  render() {
    return (
      <View style={styles.body}>
        <Image
          source={require('../img/chorum-splash.png')}
          style={styles.image}
        />
        <Text style={styles.errorTextStyle}>
          {this.props.error}
        </Text>
        <View style={styles.loginField}>
          <FontAwesomeIcon style={styles.inputIcon} icon={faAt} />
          <TextInput
            style={styles.input}
            autoCapitalize='none'
            placeholder="your@email.com"
            onChangeText={this.onEmailChange.bind(this)}
            value={this.props.email}
            keyboardType="email-address"
            autoCompleteType="email"
            textContentType="emailAddress"
          />
        </View>

        <View style={styles.loginField}>
          <FontAwesomeIcon style={styles.inputIcon} icon={faKey} />
          <TextInput
            style={styles.input}
            secureTextEntry
            placeholder="**********"
            onChangeText={this.onPasswordChange.bind(this)}
            value={this.props.password}
            textContentType="password"
            autoCompleteType="password"
          />
        </View>

        {this.renderButton()}

        <TouchableHighlight  style={styles.bottomLink} onPress={() => { Actions.register() }}>
          <Text style={styles.registerText}>Register new account</Text>
        </TouchableHighlight>
      </View>
    );
  }

  onEmailChange(text) {
    this.props.emailChanged(text);
  }

  onPasswordChange(text) {
    this.props.passwordChanged(text);
  }

  onButtonPress() {
    const { email, password } = this.props;

    this.props.loginUser({ email, password });
  }

  onRegisterLinkPress() {
    Action
  }

  renderButton() {
    if (this.props.loading) {
      return <Spinner style={{ flex: null, padding: 23 }} size="large" color={colors.green} />;
    }

    return (
      <Button
        buttonStyle={styles.loginButton}
        textStyle={styles.loginButtonText}
        onPress={this.onButtonPress.bind(this)}
      >
        Login
      </Button>
    );
  }
}


const styles = StyleSheet.create({
  body: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: colors.purple,
    justifyContent: 'flex-start'
  },
  loginField: {
    flexDirection: 'row',
    borderRadius: 10,
    paddingLeft: 10,
    paddingRight: 10,
    marginLeft: 30,
    marginRight: 30,
    marginBottom: 10,
    marginTop: 10,
    backgroundColor: colors.white,
  },
  input: {
    flex: 1,
    fontSize: 20,
    padding: 15
  },
  inputIcon: {
    color: 'grey',
    padding: 10,
    paddingRight: 20,
    alignSelf: 'center'
  },
  loginButton: {
    flex: null,
    borderRadius: 10,
    backgroundColor: colors.green,
    borderWidth: 0,
    padding: 10,
    marginLeft: 30,
    marginRight: 30,
    marginTop: 10,
    marginBottom: 10,
  },
  loginButtonText: {
    color: colors.grey,
    fontSize: 18,
    fontWeight: '600'
  },
  errorTextStyle: {
    fontSize: 20,
    alignSelf: 'center',
    color: 'red'
  },
  bottomLink: {
    alignContent: 'center',
    paddingBottom: 15
  },
  registerText: {
    fontSize: 18,
    alignSelf: 'center',
    color: colors.white,
  },
  image: {
    width: 300,
    height: 300,
    alignSelf: 'center',
  },
  smallCard: {
    paddingLeft: 30,
    paddingRight: 30
  }
})


const mapStateToProps = ({ auth }) => {
  const { email, password, error, loading } = auth;

  return { email, password, error, loading };
};

export default connect(mapStateToProps, {
  emailChanged, passwordChanged, loginUser
})(LoginForm);
