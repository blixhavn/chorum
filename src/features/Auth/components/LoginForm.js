import React, { Component } from 'react';
import { View, Text, Image, TouchableHighlight, StyleSheet, TextInput } from 'react-native';
import { connect } from 'react-redux';
import { Actions } from 'react-native-router-flux';
import { faAt, faKey } from '@fortawesome/free-solid-svg-icons';
import { emailChanged, passwordChanged, loginUser } from '../actions';
import { Spinner, Input2, Button2 } from '../../../components';
import { colors } from '../../../api/constants';

export class LoginForm extends Component {

  render() {
    return (
      <View style={styles.body}>
        <Image
          source={require('../../../assets/img/chorum-splash.png')}
          style={styles.image}
        />
        <Text style={styles.errorTextStyle}>
          {this.props.error}
        </Text>

        <Input2
          icon={faAt}
          autoCapitalize='none'
          placeholder="your@email.com"
          onChangeText={this.onEmailChange.bind(this)}
          value={this.props.email}
          keyboardType="email-address"
          autoCompleteType="email"
          textContentType="emailAddress"
        />

        <Input2
          icon={faKey}
          secureTextEntry
          placeholder="**********"
          onChangeText={this.onPasswordChange.bind(this)}
          value={this.props.password}
          textContentType="password"
          autoCompleteType="password"
        />

        {this.renderButton()}

        <TouchableHighlight  style={styles.bottomLink} onPress={() => { Actions.register() }}>
          <Text style={styles.registerText}>Register new account</Text>
        </TouchableHighlight>
      </View>
    );
  }

  onEmailChange(text) {
    this.props.emailChanged(text);
  }

  onPasswordChange(text) {
    this.props.passwordChanged(text);
  }

  onButtonPress() {
    const { email, password } = this.props;

    this.props.loginUser({ email, password });
  }

  onRegisterLinkPress() {
    Action
  }

  renderButton() {
    if (this.props.loading) {
      return <Spinner style={{ flex: null, padding: 23 }} size="large" color={colors.green} />;
    }

    return (
      <Button2
        onPress={this.onButtonPress.bind(this)}
      >
        Login
      </Button2>
    );
  }
}


const styles = StyleSheet.create({
  body: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: colors.purple,
  },
  errorTextStyle: {
    fontSize: 20,
    alignSelf: 'center',
    color: 'red'
  },
  bottomLink: {
    alignContent: 'center',
    paddingBottom: 15
  },
  registerText: {
    fontSize: 18,
    alignSelf: 'center',
    color: colors.white,
  },
  image: {
    width: 300,
    height: 300,
    alignSelf: 'center',
  }
})


const mapStateToProps = ({ auth }) => {
  const { email, password, error, loading } = auth;

  return { email, password, error, loading };
};

export default connect(mapStateToProps, {
  emailChanged, passwordChanged, loginUser
})(LoginForm);
