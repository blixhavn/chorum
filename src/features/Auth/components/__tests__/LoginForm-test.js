import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import { LoginForm } from '../LoginForm';

describe("LoginForm Component", () => {
    const props = {
      emailChanged: jest.fn(),
      passwordChanged: jest.fn(),
      loginUser: jest.fn(),
      email: 'test@email.com',
      password: 'password',
      error: '',
      loading: false
    }

    it("renders correctly", () => {
      const loginForm = renderer.create(<LoginForm {...props} />).toJSON();
      expect(loginForm).toMatchSnapshot();
    });
});
