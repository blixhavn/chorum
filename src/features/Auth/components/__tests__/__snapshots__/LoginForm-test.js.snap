// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#5C2569",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/img/chorum-splash.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 300,
        "width": 300,
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "red",
        "fontSize": 20,
      }
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 10,
        "flexDirection": "row",
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={16}
      bbWidth={16}
      fill="grey"
      focusable="false"
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "alignSelf": "center",
            "padding": 10,
            "paddingRight": 20,
          },
          null,
          Object {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={512}
      vbWidth={512}
      width={16}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4286611584,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      >
        <RNSVGPath
          d="M256 8C118.941 8 8 118.919 8 256c0 137.059 110.919 248 248 248 48.154 0 95.342-14.14 135.408-40.223 12.005-7.815 14.625-24.288 5.552-35.372l-10.177-12.433c-7.671-9.371-21.179-11.667-31.373-5.129C325.92 429.757 291.314 440 256 440c-101.458 0-184-82.542-184-184S154.542 72 256 72c100.139 0 184 57.619 184 160 0 38.786-21.093 79.742-58.17 83.693-17.349-.454-16.91-12.857-13.476-30.024l23.433-121.11C394.653 149.75 383.308 136 368.225 136h-44.981a13.518 13.518 0 0 0-13.432 11.993l-.01.092c-14.697-17.901-40.448-21.775-59.971-21.775-74.58 0-137.831 62.234-137.831 151.46 0 65.303 36.785 105.87 96 105.87 26.984 0 57.369-15.637 74.991-38.333 9.522 34.104 40.613 34.103 70.71 34.103C462.609 379.41 504 307.798 504 232 504 95.653 394.023 8 256 8zm-21.68 304.43c-22.249 0-36.07-15.623-36.07-40.771 0-44.993 30.779-72.729 58.63-72.729 22.292 0 35.601 15.241 35.601 40.77 0 45.061-33.875 72.73-58.161 72.73z"
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCompleteType="email"
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f1fa",
            "M256 8C118.941 8 8 118.919 8 256c0 137.059 110.919 248 248 248 48.154 0 95.342-14.14 135.408-40.223 12.005-7.815 14.625-24.288 5.552-35.372l-10.177-12.433c-7.671-9.371-21.179-11.667-31.373-5.129C325.92 429.757 291.314 440 256 440c-101.458 0-184-82.542-184-184S154.542 72 256 72c100.139 0 184 57.619 184 160 0 38.786-21.093 79.742-58.17 83.693-17.349-.454-16.91-12.857-13.476-30.024l23.433-121.11C394.653 149.75 383.308 136 368.225 136h-44.981a13.518 13.518 0 0 0-13.432 11.993l-.01.092c-14.697-17.901-40.448-21.775-59.971-21.775-74.58 0-137.831 62.234-137.831 151.46 0 65.303 36.785 105.87 96 105.87 26.984 0 57.369-15.637 74.991-38.333 9.522 34.104 40.613 34.103 70.71 34.103C462.609 379.41 504 307.798 504 232 504 95.653 394.023 8 256 8zm-21.68 304.43c-22.249 0-36.07-15.623-36.07-40.771 0-44.993 30.779-72.729 58.63-72.729 22.292 0 35.601 15.241 35.601 40.77 0 45.061-33.875 72.73-58.161 72.73z",
          ],
          "iconName": "at",
          "prefix": "fas",
        }
      }
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="your@email.com"
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
          "fontSize": 20,
          "lineHeight": 23,
          "padding": 15,
        }
      }
      textContentType="emailAddress"
      underlineColorAndroid="transparent"
      value="test@email.com"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 10,
        "flexDirection": "row",
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={16}
      bbWidth={16}
      fill="grey"
      focusable="false"
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "alignSelf": "center",
            "padding": 10,
            "paddingRight": 20,
          },
          null,
          Object {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={512}
      vbWidth={512}
      width={16}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4286611584,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      >
        <RNSVGPath
          d="M512 176.001C512 273.203 433.202 352 336 352c-11.22 0-22.19-1.062-32.827-3.069l-24.012 27.014A23.999 23.999 0 0 1 261.223 384H224v40c0 13.255-10.745 24-24 24h-40v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24v-78.059c0-6.365 2.529-12.47 7.029-16.971l161.802-161.802C163.108 213.814 160 195.271 160 176 160 78.798 238.797.001 335.999 0 433.488-.001 512 78.511 512 176.001zM336 128c0 26.51 21.49 48 48 48s48-21.49 48-48-21.49-48-48-48-48 21.49-48 48z"
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <TextInput
      allowFontScaling={true}
      autoCompleteType="password"
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f084",
            "M512 176.001C512 273.203 433.202 352 336 352c-11.22 0-22.19-1.062-32.827-3.069l-24.012 27.014A23.999 23.999 0 0 1 261.223 384H224v40c0 13.255-10.745 24-24 24h-40v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24v-78.059c0-6.365 2.529-12.47 7.029-16.971l161.802-161.802C163.108 213.814 160 195.271 160 176 160 78.798 238.797.001 335.999 0 433.488-.001 512 78.511 512 176.001zM336 128c0 26.51 21.49 48 48 48s48-21.49 48-48-21.49-48-48-48-48 21.49-48 48z",
          ],
          "iconName": "key",
          "prefix": "fas",
        }
      }
      onChangeText={[Function]}
      placeholder="**********"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "flex": 1,
          "fontSize": 20,
          "lineHeight": 23,
          "padding": 15,
        }
      }
      textContentType="password"
      underlineColorAndroid="transparent"
      value="password"
    />
  </View>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#A2D996",
        "borderRadius": 10,
        "borderWidth": 0,
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#27282D",
          "fontSize": 18,
          "fontWeight": "600",
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      Login
    </Text>
  </View>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "paddingBottom": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "fontSize": 18,
        }
      }
    >
      Register new account
    </Text>
  </View>
</View>
`;
